openlibrary/plugins/upstream/account.py

```diff
@@ -724,6 +726,83 @@ def get_works(self, key, page=1, limit=RESULTS_PER_PAGE):
             #works = web.ctx.site.get_many([ ... ])
             raise
 
+
+class PatronBookNotes(object):
+    """ Manages the patron's book notes and observations """
+
+    def __init__(self, user=None, limit=RESULTS_PER_PAGE, page=1):
+        self.user = user or accounts.get_current_user()
+        username = user.key.split('/')[-1]
+        self.notes_and_observations = self.get_notes_and_observations(
+            username,
+            limit=limit,
+            page=page)
+        self.notes_and_observations_count = self.get_count(username)
+
+    def get_notes_and_observations(self, username, limit=RESULTS_PER_PAGE, page=1):
+        notes_and_observations = Booknotes.get_patron_booknotes_and_observations(
+            username,
+            limit=limit,
+            page=page)
+
+        # Transform and enrich notes and observations data:
+        for entry in notes_and_observations:
+            entry['work_key'] = f"/works/OL{entry['work_id']}W"
+            entry['work'] = self.get_work(entry['work_key'])
+
+            # Convert observation type and value IDs to strings:
+            if entry['observations'] is not None:
+                ids = {}
+                for item in entry['observations']:
+                    ids[item['observation_type']] = item['observation_values']
+                entry['observations'] = convert_observation_ids(ids)
+
+            if entry['notes'] is not None:
+                entry['notes'] = {i['edition_id']: i['notes'] for i in entry['notes']}
+                entry['editions'] = {
+                    k: web.ctx.site.get(f'/books/OL{k}M')
+                    for k in entry['notes'] if k != NULL_EDITION_VALUE}
+
+        return notes_and_observations
+
+    def get_work(self, work_key):
+        work = web.ctx.site.get(work_key)
+        author_keys = [a.author.key for a in work.get('authors', [])]
+
+        return {
+            'cover_url': (
+                work.get_cover_url('S') or
```

Comment:
Is this a correct way to get a work's cover image?  I'm not able to import works locally and images for the default works seem to be fetched using IA IDs.

Reply:
I think this should get us what we want -- I'll throw on testing so we can see the behavior